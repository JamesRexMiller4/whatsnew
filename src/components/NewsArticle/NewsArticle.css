.news-card-div {
    width: 100%;
    height: fit-content;
    color: white;
    border: 10px solid;
    border-radius: 5px;
    border-color: rgb(63, 215, 164);
}

.article-image {
    width: 100%;
    height: 150px;
    border-bottom: 5px solid;
    border-color:  rgb(63, 215, 164);
}

.article-paragraph {
    width: 100%;
    height: fit-content;
    padding: 20px;
    margin-bottom: 20px;
}

.article-anchor {
    display: block;
    float: bottom;
    width: 100%;
    background-color: rgb(214, 51, 51);
    color: white;
    font-size: 25px;
    font-weight: bold;
    text-decoration: none;
}