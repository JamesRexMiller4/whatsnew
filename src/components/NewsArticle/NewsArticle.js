import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({id, headline, image, description, url}) => {
    return (
        <div className='news-card-div'>
          <img src={image} className='article-image' alt='a cat sits on a tree'/>
          <h2>{headline}</h2>
          <p className='article-paragraph'>{description}</p>
          <a className='article-anchor' href={url}>Link to Article</a>
        </div>
    )
}

export default NewsArticle;