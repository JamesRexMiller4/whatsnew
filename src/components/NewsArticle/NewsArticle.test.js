import React from 'react';
import NewsArticle from './NewsArticle';
import {shallow} from 'enzyme';

describe('NewsArticle', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<NewsArticle
            id = {2}
                key = {2}
                image = {'https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376'}
                headline = {'Giant Chicken Skeleton In Denverâ€™s Central Library Stands For More Than Just Dinner'}
                description = {'The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library. Skeletal wings extend to the side, and it looks downright prehistoric.'}
                url = {'https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/'}
            />)
    })

    it('should match the snapshot with all the data passed through', () => {
        expect(wrapper).toMatchSnapshot()
    })
})