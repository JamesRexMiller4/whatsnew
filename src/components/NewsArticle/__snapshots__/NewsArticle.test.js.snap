// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all the data passed through 1`] = `
<div
  className="news-card-div"
>
  <img
    alt="a cat sits on a tree"
    className="article-image"
    src="https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376"
  />
  <h2>
    Giant Chicken Skeleton In Denverâ€™s Central Library Stands For More Than Just Dinner
  </h2>
  <p
    className="article-paragraph"
  >
    The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library. Skeletal wings extend to the side, and it looks downright prehistoric.
  </p>
  <a
    className="article-anchor"
    href="https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
  >
    Link to Article
  </a>
</div>
`;
